src/rule-applier.js

```diff
@@ -88,46 +101,95 @@ module.exports = class RuleApplier {
   }
 
   /**
+   * Apply Rule B2: remove card from kanban.
+   *
    * @param {Context} context
    */
   async applyRuleB2 (context) {
     const issueId = context.payload.issue.number;
-
-    const getRelatedCardPromise = this.issueDataProvider.getRelatedCardInKanban(issueId);
-    const relatedCard = await getRelatedCardPromise;
+    const relatedCard = await this.issueDataProvider.getRelatedCardInKanban(issueId);
 
     this.githubApiClient.projects.deleteProjectCard({card_id: relatedCard.id});
   }
 
   /**
+   * Apply Rule C1: move card into todo column.
+   *
    * @param {Context} context
    */
   async applyRuleC1 (context) {
     const issueId = context.payload.issue.number;
-
-    const getRelatedCardPromise = this.issueDataProvider.getRelatedCardInKanban(issueId);
-    const relatedCard = await getRelatedCardPromise;
+    const relatedCard = await this.issueDataProvider.getRelatedCardInKanban(issueId);
 
     this.githubApiClient.projects.moveProjectCard({
       card_id: relatedCard.id,
       position: 'bottom',
-      column_id: this.config.kanbanColumns.toDoColumnId
+      column_id: Config.get('botConfig.kanbanColumns.todo.id')
     });
   }
 
   /**
+   * Apply Rule C2: move card into done column.
+   *
    * @param {Context} context
    */
   async applyRuleC2 (context) {
     const issueId = context.payload.issue.number;
+    const relatedCard = await this.issueDataProvider.getRelatedCardInKanban(issueId);
+
+    this.githubApiClient.projects.moveProjectCard({
+      card_id: relatedCard.id,
+      position: 'bottom',
+      column_id: Config.get('botConfig.kanbanColumns.done.id')
+    });
+  }
 
-    const getRelatedCardPromise = this.issueDataProvider.getRelatedCardInKanban(issueId);
+  /**
+   * Apply Rule E1: apply milestone to linked issues.
+   *
+   * @param {Context} context
+   */
+  async applyRuleE1 (context) {
+    const milestone = context.payload.issue.milestone.title;
+    const linkedIssueNumbers = this.getLinkedIssueNumbers(context.payload.issue);
+
+    linkedIssueNumbers.forEach((issueNumber) => {
+      this.githubApiClient.issues.edit({
+        owner: Config.get('botConfig.repository.owner'),
+        repo: Config.get('botConfig.repository.name'),
+        number: issueNumber,
+        milestone: milestone
+      });
+    });
+  }
+
+  /**
+   * Apply Rule E3: move card into to be tested column.
+   *
+   * @param {Context} context
+   */
+  async applyRuleE3 (context) {
+    // @todo: unify ways to access webhook data (issue or pull_request)
+    const issueNumber = context.payload.pull_request.number;
+
+    const getRelatedCardPromise = this.issueDataProvider.getRelatedCardInKanban(issueNumber);
     const relatedCard = await getRelatedCardPromise;
 
     this.githubApiClient.projects.moveProjectCard({
       card_id: relatedCard.id,
       position: 'bottom',
-      column_id: this.config.kanbanColumns.doneColumnId
+      column_id: Config.get('botConfig.kanbanColumns.to_be_tested.id')
```

Comment:
This string can be a global const no?

Reply:
It's kind of already a const as it defines the path to get the config item in https://github.com/PrestaShop/issuebot/pull/3/files#diff-1e9c3d615e9ebaaaa3669b4c2fd87d00R1

Like a URL :) but then I would make N constants, one for each leaf of the config tree